crates/block-producer/src/mempool/batch_graph.rs

```diff
@@ -204,23 +204,26 @@ impl BatchGraph {
     }
 
     /// Returns at most `count` batches which are ready for inclusion in a block.
-    pub fn select_block(&mut self, count: usize) -> BTreeMap<BatchJobId, TransactionBatch> {
+    pub fn select_block(
+        &mut self,
+        mut budget: BlockBudget,
+    ) -> BTreeMap<BatchJobId, TransactionBatch> {
         let mut batches = BTreeMap::new();
 
-        for _ in 0..count {
-            // This strategy just selects arbitrary roots for now. This is valid but not very
-            // interesting or efficient.
-            let Some(batch_id) = self.inner.roots().first().copied() else {
+        while let Some(batch_id) = self.inner.roots().first().copied() {
+            // SAFETY: Since it was a root batch, it must definitely have a processed batch
+            // associated with it.
+            let batch = self.inner.get(&batch_id).unwrap().clone();
```

Comment:
We have no such limitations in our coding style yet, but I think, we should avoid using `unwrap`'s in favor of `expect`. We can put safety description in panic message here. Even if it panic, the user/developer will have info why it panicked.

Reply:
I've gone back and forth on `unwrap` vs `expect` a few times. `expect` should indicate what you [expected to happen](https://doc.rust-lang.org/std/error/index.html#common-message-styles), and will mostly be shorter than the `SAFETY` comment I would write instead.

To me the two aren't quite the same i.e. will contain different strings. Or put differently, I prefer having a lengthy `SAFETY` paragraph instead of a one-line expect message.

Though maybe I should just be doing both.

What I will say is that with `expect` I often find myself writing/copying the same message in various places which sort of negates identifying the actual point-of-failure. In which case we should be running with `BACKTRACE=1` always so we get actual line and panic info. There isn't really a good reason not to do this.

---------------------------------

cc @bobbinth for his style guide input :D